src/huntsman/pocs/observatory.py

```diff
@@ -789,3 +514,127 @@ def _move_all_filterwheels_to(self, filter_name, camera_names=None):
+                required to keep the exposure, expressed as a fraction of the dynamic range.
+                Default: 0.05.
+            timeout (float): The timeout on top of the exposure time, default 60s.
+            bias (int): The bias to subtract from the frames. TODO: Use a real bias image!
+            safety_func (None or callable): If given, calls to this object return True if safe to
+                continue.
+            **kwargs: Parsed to FlatFieldSequence.
+        """
+        cam_names = list(self.cameras.keys())
+
+        # Create a flat field sequence for each camera
+        sequences = {}
+        for cam_name in cam_names:
+            target_counts, counts_tolerance = self._autoflat_target_counts(
+                cam_name, target_scaling, scaling_tolerance)
+            sequences[cam_name] = FlatFieldSequence(
+                target_counts=target_counts, counts_tolerance=counts_tolerance, bias=bias,
+                **kwargs)
+
+        # Loop until sequence has finished
+        self.logger.info(f"Starting flat field sequence for {len(cam_names)} cameras.")
+        while not all([s.is_finished for s in sequences.values()]):
+
+            if not safety_func():
+                self.logger.warning("Terminating flat fields because safety check failed.")
+                return
+
+            # Slew to field
+            self.logger.info(f'Slewing to flat field coordinates: {observation.field}.')
+            self.mount.set_target_coordinates(observation.field)
+            self.mount.slew_to_target()
+
+            # Get standard fits headers
+            headers = self.get_standard_headers(observation=observation)
+
+            # Start the exposures on each camera
+            events = {}
+            exptimes = {}
+            filenames = {}
+            start_times = {}
+            for cam_name, seq in sequences.items():
+                camera = self.cameras[cam_name]
+
+                # Get exposure time, filename and current time
+                exptimes[cam_name] = seq.get_next_exptime(past_midnight=self.past_midnight)
+                filenames[cam_name] = observation.get_exposure_filename(camera)
+                start_times[cam_name] = current_time()
+
+                # Start the exposure and get event
+                # TODO: Replace with concurrent.futures
+                events[cam_name] = camera.take_observation(
+                    observation, headers=headers, filename=filenames[cam_name],
+                    exptime=exptimes[cam_name])
+
+            # Wait for the exposures, dropping cameras that timeout
+            duration = get_quantity_value(max(exptimes.values()), u.second) + timeout
+            self._wait_for_camera_events(events, duration, remove=True)
+
+            # Update the flat field sequences with new data
+            for cam_name in list(sequences.keys()):
+                if cam_name not in self.cameras.keys():  # Camera removed
```

Comment:
Should just be able to do this, which might be safer as `keys()` doesn't return a list directly.

```suggestion
                if cam_name not in self.cameras:  # Camera removed
```